crates/ruff/src/rules/flake8_comprehensions/rules/unnecessary_map.rs

```diff
@@ -177,17 +193,25 @@ pub(crate) fn unnecessary_map(
                 return;
             }
 
-            if parameters
-                .as_ref()
-                .is_some_and(|parameters| late_binding(parameters, body))
-            {
-                return;
+            if let Some(parameters) = parameters.as_ref() {
+                if late_binding(parameters, body) {
+                    return;
+                }
+
+                parameters
+                    .posonlyargs
+                    .iter()
+                    .chain(&parameters.args)
+                    .chain(&parameters.kwonlyargs)
+                    .any(|param| param.default.is_some())
```

Comment:
Should we also avoid starred arguments and kwargs?

Reply:
Yeah, we should otherwise the fix isn't correct either:
```python
map(lambda *x: len(x), nums)

# Gets fixed to 
(len(x) for _ in nums)  # where's x?
```